File: string/count-asterisks/count-asterisks.go
package countasterisks

func countAsterisks(s string) int {
	counting := true
	asterisks := 0
	for _, c := range s {
		if c == '|' {
			if counting {
				counting = false
			} else {
				counting = true
			}
		}

		if c == '*' && counting {
			asterisks++
		}
	}

	return asterisks
}

File: string/count-asterisks/count-asterisks.ts
function countAsterisks(s: string): number {
    let asterisks = 0;
    let counting = true;

    for (let c of s) {
        if (c === '|') {
            if (counting) {
                counting = false;
                continue;
            }
            counting = true;
        }

        if (c === '*' && counting) {
            asterisks++;
        }
    }

    return asterisks;
};
